pkg/ingester-rf1/flush.go

```diff
@@ -115,10 +116,17 @@ func (i *Ingester) flushSegment(ctx context.Context, j int, w *wal.SegmentWriter
 	}
 
 	i.metrics.flushesTotal.Add(1)
-	if err := i.store.PutObject(ctx, fmt.Sprintf("loki-v2/wal/anon/"+id.String()), buf); err != nil {
+	if err := i.store.PutObject(ctx, fmt.Sprintf("loki-v2/wal/anon/"+id), buf); err != nil {
 		i.metrics.flushFailuresTotal.Inc()
 		return fmt.Errorf("store put chunk: %w", err)
 	}
 
+	if _, err := i.metastoreClient.AddBlock(ctx, &metastorepb.AddBlockRequest{
```

Comment:
Do we need a kind of 2pc here? What do we do about orphaned objects that were written to object storage but for whatever reason couldn't be added to the metastore?

Reply:
I suspect they will get cleaned up by a something else.